### PR TITLE
Fix new site with new Data

### LINKED ISSUE #220: [BUG] when a new site gets data the target displays as "undefined" in the chart page table

### PR DESCRIPTION
**Pull request by Elkrival**
* Fixed a bug when a new site is added and then data is added to that site it displays undefined
### With subject data that matches count
Screen Shot 2022-11-02 at 10 41 17 PM
### with subject data and no subject count
Screen Shot 2022-11-02 at 10 41 31 PM

### CODE REVIEW

File: server/controllers/chartsController/helpers.test.js [hunk 1]
```diff
@@ -171,6 +172,7 @@ describe('chartsController - helpers', () => {
         MA: { count: 10, targetTotal: undefined },
         NY: { count: 35, targetTotal: 35 },
         Foo: { count: 5, targetTotal: 20 },
+        empty: undefined,
```
Comment (mikestone14) [inline server/controllers/chartsController/helpers.test.js]: where is this empty study coming from? It isn't in the list of allowed studies.

File: server/controllers/chartsController/helpers.js [hunk 2]
```diff
@@ -144,12 +149,13 @@ export const postProcessData = (data, studyTotals) => {
   for (const [key, count] of data) {
     const [study, valueLabel, targetValue] = key.split('-')
     const totalsForStudy = studyTotals[study]
-    const totals = totalsForStudy.targetTotal || totalsForStudy.count
+    const totals = totalsForStudy?.targetTotal || totalsForStudy?.count
```
Comment (mikestone14) [inline server/controllers/chartsController/helpers.js]: what is the reason for this change? When would we not have totals for a study or counts for a study?

File: server/controllers/chartsController/helpers.test.js (hunk 1 above)
Comment (Elkrival) [reply server/controllers/chartsController/helpers.test.js]: That's what a new site is added, it has data but that data does not match any of the values we are looking for

File: server/controllers/chartsController/helpers.js (hunk 2 above)
Comment (Elkrival) [reply server/controllers/chartsController/helpers.js]: when we have a new site with new data these are some of the scenarios. The chart form hasn't been edited to add a target so we don't have a way to calculate it and if the new data does not have a value for a variable

File: server/controllers/chartsController/helpers.test.js [hunk 3]
```diff
@@ -106,6 +106,7 @@ describe('chartsController - helpers', () => {
   describe(helpers.generateStudyTargetTotals, () => {
     it('generates a study targets object for all sites and the totals site', () => {
       const targetValues = { MA: '10', NY: '50', Foo: '4' }
+      const allowedStudies = ['MA', 'NY', 'Foo']
```
Comment (mikestone14) [inline server/controllers/chartsController/helpers.test.js:109]:
```suggestion
      const allowedStudies = ['MA', 'NY', 'Foo', 'Bar']
```

File: server/controllers/chartsController/helpers.test.js (hunk 1 above)
Comment (mikestone14) [inline server/controllers/chartsController/helpers.test.js]:
```suggestion
        Bar: { count: 0, targetTotal: undefined },
```

Comment (mikestone14) [reply server/controllers/chartsController/helpers.test.js]: I think we can remove this. If a new site is added the study totals will look like `empty: { count: 0, targetTotal: undefined }`.

File: server/controllers/chartsController/helpers.js [hunk 4]
```diff
@@ -123,7 +123,12 @@ export const processData = ({
   }
 }
 
-export const processTotals = ({ shouldCountSubject, studyTotals, study }) => {
+export const processTotals = ({
+  shouldCountSubject,
+  studyTotals,
+  study,
+  targetValue,
```
Comment (mikestone14) [inline server/controllers/chartsController/helpers.js:130]: we are passing in the `targetValue` and setting the target value on line 150 so there is variable name collision. Do we need to pass in the target value?

Comment (Elkrival) [reply server/controllers/chartsController/helpers.js:130]: @mikestone14  we need to pass the targetValue, it is being set on line 133 https://github.com/AMP-SCZ/dpdash/blob/a42f0ed1b8e76c7426e25051ad8e25c833b25dc4/server/controllers/chartsController/helpers.js#L131-L134 but it is missing as part of the function parameters https://github.com/AMP-SCZ/dpdash/blob/a42f0ed1b8e76c7426e25051ad8e25c833b25dc4/server/controllers/chartsController/helpers.js#L126

This should be fine the targetValue on line 150 is part of a different scope.

https://github.com/AMP-SCZ/dpdash/blob/a42f0ed1b8e76c7426e25051ad8e25c833b25dc4/server/controllers/chartsController/helpers.js#L126-L145

Comment (mikestone14) [reply server/controllers/chartsController/helpers.js:130]: ~~@Elkrival if we're passing the target value in here do we still need to get the target value from the `key` when we iterate through the `data`?~~

nevermind! I think I was confusing the diff of 2 different functions. 👍

File: server/controllers/chartsController/helpers.test.js [hunk 5]
```diff
@@ -121,13 +122,39 @@ describe('chartsController - helpers', () => {
...
 
-      expect(helpers.generateStudyTargetTotals(chart)).toEqual({
+      expect(helpers.generateStudyTargetTotals(chart, allowedStudies)).toEqual({
         MA: { count: 0, targetTotal: 20 },
         NY: { count: 0, targetTotal: 100 },
         Foo: { count: 0, targetTotal: 8 },
         Totals: { count: 0, targetTotal: 128 },
       })
     })
+    it('results with a targetTotal of undefined when a site does not include targetValues', () => {
```
Comment (mikestone14) [inline server/controllers/chartsController/helpers.test.js]: this is a confusing test description. Is this testing that the "totals" site will have an undefined targetTotal when a site does not have a target value?

File: server/controllers/chartsController/helpers.test.js [hunk 6]
```diff
@@ -229,5 +256,76 @@ describe('chartsController - helpers', () => {
         },
       })
     })
+    it('builds an object with graph data per site when there are only counts and no targets', () => {
```
Comment (mikestone14) [inline server/controllers/chartsController/helpers.test.js]:
```suggestion
    })
  
    it('builds an object with graph data per site when there are only counts and no targets', () => {
```

File: server/controllers/chartsController/helpers.js (hunk 4 above)
Comment (mikestone14) [reply server/controllers/chartsController/helpers.js:130]: it seems like we need a test for this helper function.

File: server/controllers/chartsController/helpers.test.js (hunk 5 above)
Comment (mikestone14) [inline server/controllers/chartsController/helpers.test.js]:
```suggestion
    })

    it('results with a targetTotal of undefined when a site does not include targetValues', () => {
```

File: server/controllers/chartsController/helpers.test.js [hunk 7]
```diff
@@ -229,5 +256,76 @@ describe('chartsController - helpers', () => {
...
+        ['NY-female-undefined', 25],
+        ['Foo-male-undefined', 5],
+        ['Foo-female-undefined', 0],
+        ['Bar-male-undefined', 0],
+        ['Bar-female-undefined', 0],
+        [`${helpers.TOTALS_STUDY}-male`, 20],
+        [`${helpers.TOTALS_STUDY}-female`, 30],
+      ])
+      const postProcessedData = helpers.postProcessData(data, studyTotals)
+      expect(Object.fromEntries(postProcessedData)).toEqual({
```
Comment (mikestone14) [inline server/controllers/chartsController/helpers.test.js]:
```suggestion
      const postProcessedData = helpers.postProcessData(data, studyTotals)

      expect(Object.fromEntries(postProcessedData)).toEqual({
```

File: server/controllers/chartsController/helpers.js (hunk 4 above)
Comment (Elkrival) [reply server/controllers/chartsController/helpers.js:130]: https://github.com/AMP-SCZ/dpdash/pull/221/commits/97d6295d9cb36f821ae3ee45edcf0d501ed06a80

File: server/controllers/chartsController/helpers.test.js (hunk 5 above)
Comment (Elkrival) [reply server/controllers/chartsController/helpers.test.js]: https://github.com/AMP-SCZ/dpdash/pull/221/commits/97d6295d9cb36f821ae3ee45edcf0d501ed06a80

File: server/controllers/chartsController/helpers.test.js (hunk 6 above)
Comment (Elkrival) [reply server/controllers/chartsController/helpers.test.js]: https://github.com/AMP-SCZ/dpdash/pull/221/commits/97d6295d9cb36f821ae3ee45edcf0d501ed06a80

File: server/controllers/chartsController/helpers.test.js (hunk 5 above)
Comment (Elkrival) [reply server/controllers/chartsController/helpers.test.js]: https://github.com/AMP-SCZ/dpdash/pull/221/commits/97d6295d9cb36f821ae3ee45edcf0d501ed06a80

File: server/controllers/chartsController/helpers.test.js (hunk 7 above)
Comment (Elkrival) [reply server/controllers/chartsController/helpers.test.js]: https://github.com/AMP-SCZ/dpdash/pull/221/commits/97d6295d9cb36f821ae3ee45edcf0d501ed06a80

**Review by mikestone14 — APPROVED** [review]
